pkg/ingester/ingester.go

```diff
@@ -162,7 +173,7 @@ func (i *Ingester) Push(ctx context.Context, req *logproto.PushRequest) (*logpro
 
 func (i *Ingester) getOrCreateInstance(instanceID string) *instance {
 	inst, ok := i.getInstanceByID(instanceID)
-	if ok {
+	if ok || i.readonly {
```

Comment:
This means `inst` can be nil ?

Reply:
hmm, good catch. It should be fine for us to create an instance as long as we don't write anything to it. I'll remove the readonly check here.